app/src/main/java/com/iti/android/team1/ebuy/ui/login_screen/view/LoginScreen.kt

```diff
@@ -27,32 +33,65 @@ class LoginScreen : Fragment() {
         savedInstanceState: Bundle?,
     ): View {
         binding = FragmentLoginScreenBinding.inflate(inflater, container, false)
+
+        if (viewModel.getAuthStateFromPrefs())
+            navigateToMainActivity()
+
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
         binding.btnSignUp.setOnClickListener {
             findNavController().navigate(
                 LoginScreenDirections.actionLoginScreen2ToRegisterScreen2()
             )
         }
 
         binding.btnLogin.setOnClickListener {
-           /* if (AuthRegex.isEmailValid(binding.edtEmail.editableText.toString()) &&
+            if (AuthRegex.isEmailValid(binding.edtEmail.editableText.toString()) &&
                 AuthRegex.isPasswordValid(binding.edtPassword.editableText.toString())
             ) {
-                ///TODO: make login call
-
+                viewModel.makeLoginRequest(
+                    email = binding.edtEmail.text.toString().trim(),
+                    password = binding.edtPassword.text.toString().trim()
+                )
+                fetchUserData()
             } else if (!AuthRegex.isEmailValid(binding.edtEmail.editableText.toString())) {
-                ///TODO: email is week
                 binding.textInputLayout.error = "Email is not valid"
             } else if (!AuthRegex.isPasswordValid(binding.edtPassword.editableText.toString())) {
-                ///TODO: password not valid
-                binding.textInputLayout2.error = "Password is not valid"
+                binding.textInputLayout2.error = "Wrong password"
```

Comment:
The logic shouldn't be in the view, Use The same sealed class as in Register Screen and put the logic on viewmodel

Reply:
okay